modules/core/04-channel/keeper/packet.go

```diff
@@ -342,14 +342,15 @@ func (k Keeper) WriteAcknowledgement(
 		return types.ErrAcknowledgementExists
 	}
 
-	if len(acknowledgement) == 0 {
+	bz := acknowledgement.Acknowledgement()
+	if len(bz) == 0 {
 		return sdkerrors.Wrap(types.ErrInvalidAcknowledgement, "acknowledgement cannot be empty")
```

Comment:
Just change this check to `if acknowledgement == nil`

Reply:
So this is what I had originally. 

Can we go with this @damiannolan @colin-axner?